config/ompi_setup_mpi_fortran.m4

```diff
@@ -376,6 +377,20 @@ end program]])],
           [OMPI_FORTRAN_BUILD_SIZEOF=0])
     AC_SUBST(OMPI_FORTRAN_BUILD_SIZEOF)
 
+    AC_ARG_ENABLE([deprecate-mpif-h],
+                  [AS_HELP_STRING([--enable-deprecate-mpif-h],
+                                  [Mark the mpif.h bindings as deprecated (default: enabled)])])
+    AS_IF([! test x"$enable_deprecate_mpif_h" = "xno"],
+            [OMPI_FORTRAN_CHECK_WARNING([enable_deprecate_mpif_h="yes"],
+                                        [enable_deprecate_mpif_h="no"])])
```

Comment:
If a human asks for something -- e.g., `--enable-deprecate-mpif-h` -- and we can't provide it (i.e., the compiler doesn't support it), we need to `AC_MSG_ERROR` to emit an error message and then abort.

Reply:
Fixed, now bailing out if the the user requested it and we cannot provide a proper warning.